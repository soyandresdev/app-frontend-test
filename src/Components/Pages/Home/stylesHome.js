import styled from "styled-components";
import { device } from ".././../../styles/mediaQuery";

export const Container = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Title = styled.h1`
  color: #c1c1c1;
`;

export const Columns = styled.div`
  display: grid;
  grid-template-columns: 1fr;
  grid-gap: 20px;
  @media ${device.tablet} {
    grid-template-columns: repeat(${({ columns }) => columns}, 1fr);
  }
`;
