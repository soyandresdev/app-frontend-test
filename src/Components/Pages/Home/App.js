import React from "react";
import Input from "../../Atoms/Input";
import Select from "../../Atoms/Select";
import Button from "../../Atoms/Button";
import BoxWhiteRound from "../../Atoms/BoxWhiteRound";
import PrivateRoute from "../../Templates/PrivateRoute";
import { useFormik } from "formik";
import * as Yup from "yup";
import { Container, Title, Columns } from "./stylesHome";
import DirectoryPerson from "../../../utils/Form/directoryPerson";

function App() {
  const formik = useFormik({
    initialValues: {
      firstName: "",
      lastName: "",
      typeDocument: "",
      job: "",
      id: "",
      city: "",
      email: "",
      addres: "",
      phone: "",
      phoneHome: "",
      dateExpedition: "",
      dateBirth: ""
    },
    validationSchema: Yup.object({
      firstName: Yup.string()
        .max(20, "Debe tener 20 caracteres o menos")
        .required("Rellena este campo obligatorio."),
      lastName: Yup.string()
        .max(20, "Debe tener 20 caracteres o menos")
        .required("Rellena este campo obligatorio."),
      job: Yup.string()
        .max(15, "Debe tener 15 caracteres o menos")
        .required("Rellena este campo obligatorio."),
      typeDocument: Yup.string().required(
        "Selecciona una opción del menú desplegable."
      ),
      id: Yup.string().required("Rellena este campo obligatorio."),
      city: Yup.string().required(
        "Selecciona una opción del menú desplegable."
      ),
      email: Yup.string()
        .required("Rellena este campo obligatorio.")
        .email("Correo electrónico no válida"),
      addres: Yup.string().required("Rellena este campo obligatorio."),
      phone: Yup.string().required("Rellena este campo obligatorio."),
      dateExpedition: Yup.string().required(
        "Selecciona una opción del menú desplegable."
      ),
      dateBirth: Yup.string().required(
        "Selecciona una opción del menú desplegable."
      )
    }),
    onSubmit: values => {
      DirectoryPerson({ person: values });
      formik.resetForm({});
    }
  });

  return (
    <PrivateRoute>
      <Container>
        <BoxWhiteRound>
          <header>
            <Title>Formulario</Title>
            <p>
              Personas naturales con las cuales podría hacer negocios en el
              futuro.
            </p>
          </header>
          <form onSubmit={formik.handleSubmit}>
            <Columns columns={2}>
              <Input
                name="firstName"
                label="Nombres"
                type="text"
                required
                onChange={formik.handleChange}
                touched={formik.touched}
                value={formik.values.firstName}
                errors={formik.errors}
              />
              <Input
                name="lastName"
                label="Apellidos"
                type="text"
                required
                onChange={formik.handleChange}
                touched={formik.touched}
                value={formik.values.lastName}
                errors={formik.errors}
              />
            </Columns>
            <Columns columns={2}>
              <Input
                name="dateBirth"
                label="Fecha de Nacimiento"
                type="date"
                required
                onChange={formik.handleChange}
                touched={formik.touched}
                value={formik.values.dateBirth}
                errors={formik.errors}
              />
              <Input
                name="job"
                label="Ocupación"
                type="text"
                required
                onChange={formik.handleChange}
                touched={formik.touched}
                value={formik.values.job}
                errors={formik.errors}
              />
            </Columns>
            <Columns columns={2}>
              <Select
                name="typeDocument"
                label="Tipo de Documento"
                required
                onChange={formik.handleChange}
                touched={formik.touched}
                value={formik.values.typeDocument}
                errors={formik.errors}
              >
                <option disabled defaultValue value="">
                  Seleccionar
                </option>
                <option value="CC">Cédula de ciudadanía</option>
                <option value="CE">Cédula de extranjería</option>
                <option value="PA">Pasaporte</option>
              </Select>
              <Input
                name="id"
                label="Número de documento"
                type="text"
                required
                onChange={formik.handleChange}
                touched={formik.touched}
                value={formik.values.id}
                errors={formik.errors}
              />
            </Columns>
            <Columns columns={2}>
              <Select
                name="city"
                label="Ciudad"
                required
                onChange={formik.handleChange}
                touched={formik.touched}
                value={formik.values.city}
                errors={formik.errors}
              >
                <option disabled defaultValue value="">
                  Seleccionar
                </option>
                <option value="Armenia">Armenia</option>
                <option value="Barranquilla">Barranquilla</option>
                <option value="Bello">Bello</option>
                <option value="Bogotá">Bogotá</option>
                <option value="Bucaramanga">Bucaramanga</option>
                <option value="Buenaventura">Buenaventura</option>
                <option value="Cali">Cali</option>
                <option value="Cartagena">Cartagena</option>
                <option value="Cúcuta">Cúcuta</option>
                <option value="Ibagué">Ibagué</option>
                <option value="Ipiales">Ipiales</option>
                <option value="Manizales">Manizales</option>
                <option value="Medellín">Medellín</option>
                <option value="Montería">Montería</option>
                <option value="Neiva">Neiva</option>
                <option value="Pasto">Pasto</option>
                <option value="Pereira">Pereira</option>
                <option value="Popayán">Popayán</option>
                <option value="Santa Marta">Santa Marta</option>
                <option value="Soacha">Soacha</option>
                <option value="Soledad">Soledad</option>
                <option value="Tuluá">Tuluá</option>
                <option value="Valledupar">Valledupar</option>
                <option value="Villavicencio">Villavicencio</option>
                <option value="Otra">Otra</option>
              </Select>
              <Input
                name="dateExpedition"
                label="Fecha de expedición"
                type="date"
                required
                onChange={formik.handleChange}
                touched={formik.touched}
                value={formik.values.dateExpedition}
                errors={formik.errors}
              />
            </Columns>
            <Columns columns={2}>
              <Input
                name="email"
                label="Correo electrónico"
                type="text"
                required
                onChange={formik.handleChange}
                touched={formik.touched}
                value={formik.values.email}
                errors={formik.errors}
              />
              <Input
                name="addres"
                label="Dirección"
                type="text"
                required
                onChange={formik.handleChange}
                touched={formik.touched}
                value={formik.values.addres}
                errors={formik.errors}
              />
            </Columns>
            <Columns columns={2}>
              <Input
                name="phone"
                label="Teléfono celular"
                type="text"
                required
                onChange={formik.handleChange}
                touched={formik.touched}
                value={formik.values.phone}
                errors={formik.errors}
              />
              <Input
                name="phoneHome"
                label="Teléfono fijo"
                type="text"
                onChange={formik.handleChange}
                touched={formik.touched}
                value={formik.values.phoneHome}
                errors={formik.errors}
              />
            </Columns>
            <Button id="save" type="submit">
              Enviar
            </Button>
          </form>
        </BoxWhiteRound>
      </Container>
    </PrivateRoute>
  );
}

export default App;
