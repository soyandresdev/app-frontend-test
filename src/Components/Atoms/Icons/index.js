import React from "react";
import PropTypes from "prop-types";

import Home from "./IconComponets/Home";
import User from "./IconComponets/User";
import AddUser from "./IconComponets/AddUser";
import Graph from "./IconComponets/Graph";
import Logout from "./IconComponets/Logout";
import Setting from "./IconComponets/Setting";
import Notification from "./IconComponets/Notification";

export default function Icons({ type, ...resProps }) {
  return (
    <>
      {{
        home: <Home {...resProps} />,
        user: <User {...resProps} />,
        addUser: <AddUser {...resProps} />,
        graph: <Graph {...resProps} />,
        logout: <Logout {...resProps} />,
        setting: <Setting {...resProps} />,
        notification: <Notification {...resProps} />
      }[`${type}`] || <div />}
    </>
  );
}

Icons.propTypes = {
  type: PropTypes.oneOf([
    "home",
    "user",
    "addUser",
    "graph",
    "logout",
    "setting",
    "notification"
  ]).isRequired
};
