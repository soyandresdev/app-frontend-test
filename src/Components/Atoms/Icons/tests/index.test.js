import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Icons from "../index";
import Home from "../IconComponets/Home";

describe("Test component [Icons]", () => {
  const Props = {
    type: "home"
  };
  const wrapper = shallow(<Icons {...Props} />);

  it("Renders correctly defaultProps [Icons]", () => {
    const rendered = renderer.create(<Icons {...Props} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it("Renders correctly [Icons]", () => {
    const rendered = renderer.create(<Icons {...Props} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it("Test init props required [Icons]", () => {
    expect(wrapper.find(Home).length).toBe(1);
  });
});
