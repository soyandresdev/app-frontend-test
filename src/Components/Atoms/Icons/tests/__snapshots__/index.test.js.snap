// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component [Icons] Renders correctly [Icons] 1`] = `
<svg
  height="24"
  viewBox="0 0 24 24"
  width="24"
>
  <rect
    fill="none"
    height="24"
    rx="0"
    ry="0"
    width="24"
  />
  <path
    clipRule="evenodd"
    d="M15.8 2C15.3582 2 15 2.35817 15 2.8V5.1L12.2867 3.20071C12.1146 3.0802 11.8854 3.0802 11.7133 3.20071L2.21327 9.85071C2.07961 9.94428 2 10.0972 2 10.2603V11.0397C2 11.4442 2.45534 11.6813 2.78673 11.4493L5 9.9V21.2C5 21.6418 5.35817 22 5.8 22H18.2C18.6418 22 19 21.6418 19 21.2V9.9L21.2133 11.4493C21.5447 11.6813 22 11.4442 22 11.0397V10.2603C22 10.0972 21.9204 9.94428 21.7867 9.85071L19 7.9V2.8C19 2.35817 18.6418 2 18.2 2H15.8ZM17.8 7.06V3.6C17.8 3.37909 17.6209 3.2 17.4 3.2H16.6C16.3791 3.2 16.2 3.37909 16.2 3.6V5.94L17.8 7.06ZM17.8 9.06V20.4C17.8 20.6209 17.6209 20.8 17.4 20.8H6.6C6.37909 20.8 6.2 20.6209 6.2 20.4V9.06L11.7133 5.20071C11.8854 5.0802 12.1146 5.0802 12.2867 5.20071L17.8 9.06Z"
    fill="#ffffff"
    fillRule="evenodd"
  />
</svg>
`;

exports[`Test component [Icons] Renders correctly defaultProps [Icons] 1`] = `
<svg
  height="24"
  viewBox="0 0 24 24"
  width="24"
>
  <rect
    fill="none"
    height="24"
    rx="0"
    ry="0"
    width="24"
  />
  <path
    clipRule="evenodd"
    d="M15.8 2C15.3582 2 15 2.35817 15 2.8V5.1L12.2867 3.20071C12.1146 3.0802 11.8854 3.0802 11.7133 3.20071L2.21327 9.85071C2.07961 9.94428 2 10.0972 2 10.2603V11.0397C2 11.4442 2.45534 11.6813 2.78673 11.4493L5 9.9V21.2C5 21.6418 5.35817 22 5.8 22H18.2C18.6418 22 19 21.6418 19 21.2V9.9L21.2133 11.4493C21.5447 11.6813 22 11.4442 22 11.0397V10.2603C22 10.0972 21.9204 9.94428 21.7867 9.85071L19 7.9V2.8C19 2.35817 18.6418 2 18.2 2H15.8ZM17.8 7.06V3.6C17.8 3.37909 17.6209 3.2 17.4 3.2H16.6C16.3791 3.2 16.2 3.37909 16.2 3.6V5.94L17.8 7.06ZM17.8 9.06V20.4C17.8 20.6209 17.6209 20.8 17.4 20.8H6.6C6.37909 20.8 6.2 20.6209 6.2 20.4V9.06L11.7133 5.20071C11.8854 5.0802 12.1146 5.0802 12.2867 5.20071L17.8 9.06Z"
    fill="#ffffff"
    fillRule="evenodd"
  />
</svg>
`;
